src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -893,6 +916,60 @@ def _check_query_cardinality(self, new_queries: Sequence[DashboardWidgetQuery]):
                 current_widget_specs,
             )
 
+    def _update_or_create_field_links(
+        self,
+        query: DashboardWidgetQuery,
+        linked_dashboards: list[LinkedDashboard],
+        widget: DashboardWidget,
+    ):
+        """
+        Update DashboardFieldLink entries for a query.
+        linked_dashboards is expected to be an array of dicts with format {"field": str, "dashboard_id": int}
+        This can be further optimized, currently we are doing one bulk update for every widget query, but we could do one bulk update for all widget queries at once.
+        In practice a table typically has only one query, so this is not a big deal.
+        """
+
+        organization = self.context["organization"]
+        if not features.has("organizations:dashboards-drilldown-flow", organization):
+            return
+
+        # Get the set of fields that should exist
+        new_fields = set()
+        field_links_to_create = []
+
+        widget_display_type = widget.display_type
+
+        if widget_display_type is not DashboardWidgetDisplayTypes.TABLE:
+            raise serializers.ValidationError("Field links are only supported for table widgets")
+
+        if linked_dashboards:
```

Comment:
This check is redundant, I think, since `linked_dashboards` is guaranteed to be  a list by the typing